src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
```

Comment:
The getPosts, getPostbyId, and getAllPosts functions make requests to the API and may throw errors. Therefore, they need to be placed in a try catch block.